src/lib/components/EventTime/index.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * @fileoverview Element that renders a time in local timezone.
+ */
+
+import {html} from 'lit-element';
+import {BaseElement} from '../BaseElement';
+import './_styles.scss';
+
+const months = [
+  'Jan',
+  'Feb',
+  'Mar',
+  'Apr',
+  'May',
+  'Jun',
+  'Jul',
+  'Aug',
+  'Sep',
+  'Oct',
+  'Nov',
+  'Dec',
+];
+
+const connected = new Set();
+let timezoneCheckInterval = 0;
+let previous = 0;
+
+const checkTimezone = () => {
+  const now = new Date().getTimezoneOffset();
+  if (now !== previous) {
+    previous = now;
+    connected.forEach((node) => node.requestUpdate());
+  }
+};
+
+/**
+ * Renders an event's time in the user's local timezone, including optional duration.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class EventTime extends BaseElement {
+  static get properties() {
+    return {
+      datetime: {type: String},
+      duration: {type: Number},
+    };
+  }
+
+  constructor() {
+    super();
+    this._date = null;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (connected.size === 0) {
+      // Every 10 minutes, check if we're in the same timezone.
+      timezoneCheckInterval = window.setInterval(checkTimezone, 10 * 60 * 1000);
+    }
+    connected.add(this);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    connected.delete(this);
+    if (connected.size === 0) {
+      window.clearInterval(timezoneCheckInterval);
+    }
+  }
+
+  shouldUpdate(changedProperties) {
+    if (changedProperties.has('datetime')) {
+      this._date = null;
+
+      if (this.datetime) {
+        const d = new Date(Date.parse(this.datetime));
+        if (+d) {
+          this._date = d;
+        }
+      }
+    }
+
+    return super.shouldUpdate(changedProperties);
+  }
+
+  render() {
+    if (!this._date) {
+      return html`
+        <time>?</time>
+      `;
+    }
+
+    const format = (d, options) => {
+      return new Intl.DateTimeFormat('default', options).format(d);
+    };
+    const options = new Intl.DateTimeFormat().resolvedOptions();
+
+    let durationPart = '';
+    if (this.duration > 0) {
+      const end = new Date(this._date);
+      end.setHours(end.getHours() + this.duration);
+      durationPart = html`
+        &mdash; ${format(end, {hour: 'numeric', minute: '2-digit'})}
+      `;
+    }
+
+    return html`
+      <time datetime=${this._date.toISOString()}>
+        <div class="et-date-feat">
```

Comment:
it seems like you could get rid of the `-feat` suffix on all of these classes.

Reply:
I'm kind of wondering if we should also just drop the `et-` prefix and adopt a policy where we use one level of CSS nesting for custom elements that use light dom. Might make the html and css tidier.

```scss
web-event-time {
  .date {}
  .date__month {}
  .date__day {}
  .time {}
  .time__value {}
  .time__timezone {}
}
```

Reply:
I've done this. I was worried that this would collide with other generic styles, but I guess the point is that we generally don't have those (we only have `.w_...`), so it is safe to use simple names when nested.

Reply:
(This is especially try as we're not reparenting children)